utils/add_platform_rule.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/python3
+
+import argparse
+import subprocess
+import pathlib
+import sys
+import os
+import time
+
+'''
+ This script is intended to help content writers create a new application check for Kubernetes.
+ The 'create' subcommand creates the initial files for a new rule and fetches the raw URL of the object in question (unless you specify the URL).
+ The 'cluster-test' subcommand pushes the content to your cluster, and then runs a Platform scan for your rule with compliance-operator.
+ The 'test' subcommand builds your content locally and tests directly using an openscap podman container.
+
+ $ utils/add_platform_rule.py create --rule=ocp_proxy_has_ca --type="proxies.config" --name="cluster" --yamlpath=".spec.trustedCA.name" --match="[a-zA-Z0-9]*"
+ creating check for "/apis/config.openshift.io/v1/proxies/cluster" with yamlpath ".spec.trustedCA.name" satisfying match of "[a-zA-Z0-9]*"
+ wrote applications/openshift/ocp_proxy_has_ca/rule.yml
+ wrote applications/openshift/ocp_proxy_has_ca/oval/shared.xml
+
+ $ utils/add_platform_rule.py cluster-test --rule=ocp_proxy_has_ca
+ testing rule ocp_proxy_has_ca in-cluster
+ deploying compliance-operator
+ pushing image build to cluster
+ waiting for cleanup from previous test run
+ output from last phase check: LAUNCHING NOT-AVAILABLE
+ output from last phase check: RUNNING NOT-AVAILABLE
+ output from last phase check: AGGREGATING NOT-AVAILABLE
+ output from last phase check: DONE COMPLIANT
+ COMPLIANT
+'''
+
+PLATFORM_RULE_DIR = 'applications/openshift'
+OSCAP_TEST_IMAGE = 'quay.io/compliance-operator/openscap-ocp:1.3.3'
+OSCAP_CMD_OPTS = 'oscap xccdf eval --fetch-remote-resources --profile xccdf_org.ssgproject.content_profile_test --results-arf /tmp/report-arf.xml /content/ssg-ocp4-ds.xml'
+
+RULE_TEMPLATE = ('''prodtype: ocp4
+
+title: {TITLE}
+
+description: TBD
+
+rationale: TBD
+
+identifiers:
+    cce@ocp4: 84209-6
+
+severity: medium
+
+warnings:
+    - general: |-
+        {{{{{{ openshift_cluster_setting("{URL}") | indent(8) }}}}}}
+''')
+
+
+OVAL_TEMPLATE = ('''{{{{% set YAML_TEST_OVAL_VERSION = [5, 11] %}}}}
+
+{{{{% if target_oval_version >= YAML_TEST_OVAL_VERSION %}}}}
+<def-group>
+      <definition class="compliance" version="1" id="{{{{{{ rule_id }}}}}}">
+      <metadata>
+        <title>{TITLE}</title>
+        {{{{{{- oval_affected(products) }}}}}}
+        <description>{DESC}</description>
+      </metadata>
+      <criteria operator="AND">
+        <criterion comment="{DESC}" negate="{NEGATE}" test_ref="test_{{{{{{ rule_id }}}}}}"/>
```

Comment:
This is causing the build to fail, because the Python boolean gets converted to True/False. My crude fix was:
```
@@ -168,7 +169,7 @@ def createFunc(args):
     print('wrote ' + rule_yaml_path)
     with open(shared_xml_path, 'w') as file:
         file.write(OVAL_TEMPLATE.format(TITLE=args.title, DESC=args.description,
-                                        URL=url, YAMLPATH=args.yamlpath, MATCH=args.match, NEGATE=args.negate))
+                                        URL=url, YAMLPATH=args.yamlpath, MATCH=args.match, NEGATE=str(args.negate).lower()))
```